google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITPitrUpdateDatabaseTest.java

```diff
@@ -0,0 +1,195 @@
+  private static DatabaseClient dbClient;
+  private static String instanceId;
+  private static String databaseId;
+
+  @BeforeClass
+  public static void setUp() throws Exception {
+    assumeFalse("PITR-lite features are not supported by the emulator", isUsingEmulator());
+
+    final RemoteSpannerHelper testHelper = env.getTestHelper();
+    final String projectId = testHelper.getOptions().getProjectId();
+    instanceId = testHelper.getInstanceId().getInstance();
+    databaseId = testHelper.getUniqueDatabaseId();
+    dbAdminClient = testHelper.getClient().getDatabaseAdminClient();
+
+    createDatabase(dbAdminClient, instanceId, databaseId, Collections.<String>emptyList());
+    updateVersionRetentionPeriod(dbAdminClient, instanceId, databaseId, VERSION_RETENTION_PERIOD);
+
+    dbClient =
+        testHelper.getClient().getDatabaseClient(DatabaseId.of(projectId, instanceId, databaseId));
+  }
+
+  @AfterClass
+  public static void tearDown() {
+    if (!isUsingEmulator()) {
+      dbAdminClient.dropDatabase(instanceId, databaseId);
+    }
+  }
+
+  @Test
+  public void returnsTheVersionRetentionPeriodSetThroughGetDatabase() {
+    final Database database = dbAdminClient.getDatabase(instanceId, databaseId);
+
+    assertThat(database.getVersionRetentionPeriod()).isEqualTo(VERSION_RETENTION_PERIOD);
+    assertThat(database.getEarliestVersionTime()).isNotNull();
+  }
+
+  @Test
+  public void returnsTheVersionRetentionPeriodSetThroughListDatabases() {
+    final Page<Database> page = dbAdminClient.listDatabases(instanceId);
+
+    for (Database database : page.iterateAll()) {
+      if (!database.getId().getDatabase().equals(databaseId)) {
+        continue;
+      }
+      assertThat(database.getVersionRetentionPeriod()).isEqualTo(VERSION_RETENTION_PERIOD);
+      assertThat(database.getEarliestVersionTime()).isNotNull();
+    }
+  }
+
+  @Test
+  public void returnsTheVersionRetentionPeriodSetThroughGetDatabaseDdl() {
+    final List<String> ddls = dbAdminClient.getDatabaseDdl(instanceId, databaseId);
+
+    boolean hasVersionRetentionPeriodStatement = false;
+    for (String ddl : ddls) {
+      hasVersionRetentionPeriodStatement =
+          ddl.contains("version_retention_period = '" + VERSION_RETENTION_PERIOD + "'");
+      if (hasVersionRetentionPeriodStatement) {
+        break;
+      }
+    }
+    assertThat(hasVersionRetentionPeriodStatement).isTrue();
+  }
+
+  @Test
+  public void returnsTheVersionRetentionPeriodSetThroughInformationSchema() {
+    final ResultSet rs =
+        dbClient
+            .singleUse()
+            .executeQuery(
+                Statement.of(
+                    "SELECT OPTION_VALUE AS version_retention_period "
+                        + "FROM INFORMATION_SCHEMA.DATABASE_OPTIONS "
+                        + "WHERE SCHEMA_NAME = '' AND OPTION_NAME = 'version_retention_period'"));
+
+    String versionRetentionPeriod = null;
+    while (rs.next()) {
```

Comment:
nit: The `ResultSet` is created using a single-use read context and it is fully consumed, so it will be closed, but I would say that it is good practice to always close the `ResultSet` when you are done with it.

Reply:
Good point, added the `close` call.